tldw_Server_API/app/core/DB_Management/AuthNZ_Metering_Repository.py

```
@@ -0,0 +1,404 @@
+class DuplicateActiveSubscriptionError(RuntimeError):
+    """Raised when a user resolves to more than one active billing subscription."""
+
+
+class _AuthNZMeteringRepositoryBase:
+    """Shared pool-loading helpers for AuthNZ metering repositories."""
+
+    def __init__(self, *, db_pool: Any | None = None) -> None:
+        """Store an optional injected pool for tests or alternate composition."""
+        self._db_pool = db_pool
+
+    async def _get_db_pool(self) -> Any:
```

Ruff (0.15.6)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `_get_db_pool`

(ANN401)